File: go/src/sum/sum.go
package sum

// https://github.com/joncalhoun/algorithmswithgo.com/tree/master/module01#06---sum-all-the-numbers-in-a-list-code

// Iterate :
func Iterate(numbers []int) int {
	result := 0
	for _, n := range numbers {
		result += n
	}
	return result
}

// Recursive :
func Recursive(numbers []int) int {
	if len(numbers) == 0 {
		return 0
	}
	return numbers[0] + Recursive(numbers[1:])
}

File: elixir/lib/polyglot/sum.ex
# https://github.com/joncalhoun/algorithmswithgo.com/tree/master/module01#06---sum-all-the-numbers-in-a-list-code

defmodule Polyglot.Sum do
  def pattern_match([]), do: 0

  def pattern_match(list) do
    [head | tail] = list
    head + pattern_match(tail)
  end

  def std(list) do
    Enum.sum(list)
  end
end
